app/src/main/kotlin/com/ivanovsky/passnotes/presentation/groups/GroupsViewModel.kt

```
@@ -599,7 +599,11 @@ class GroupsViewModel(
                 return@launch
             }
 
-            val password = (getKeyResult.obj as PasswordKeepassKey).password
+            val password = when (val key = getKeyResult.getOrThrow()) {
+                is PasswordKeepassKey -> key.password
+                is FileKeepassKey -> key.password
+                else -> throw IllegalStateException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)